endpoint/endpoint.go

```diff
@@ -33,9 +36,17 @@ type Endpoint struct {
 
 // NewEndpoint initialization method to be used to create an endpoint
 func NewEndpoint(dnsName, target string) *Endpoint {
+	if net.ParseIP(target) == nil {
+		target = sanitizeHostname(target)
+	}
 	return &Endpoint{
-		DNSName: dnsName,
+		DNSName: sanitizeHostname(dnsName),
 		Target:  target,
 		Labels:  map[string]string{},
 	}
 }
+
+// sanitizeHostname appends a trailing dot to a hostname if it's missing.
+func sanitizeHostname(hostname string) string {
```

Comment:
@linki I think I will just remove this function, seems unnecessary to me right now , will also change to trim "." from target, and sanitize dns name